src/createAnimatedComponent/InlinePropManager.ts

```diff
@@ -0,0 +1,190 @@
+  >
+): Record<string, unknown> {
+  const inlineProps: Record<string, unknown> = {};
+
+  for (const key in props) {
+    const value = props[key];
+    if (key === 'style') {
+      const styles = flattenArray<StyleProps>(props.style ?? []);
+      styles.forEach((style) => {
+        if (!style) {
+          return;
+        }
+        for (const [key, styleValue] of Object.entries(style)) {
+          if (isSharedValue(styleValue)) {
+            inlineProps[key] = styleValue;
+          } else if (
+            key === 'transform' &&
+            isInlineStyleTransform(styleValue)
+          ) {
+            inlineProps[key] = styleValue;
+          }
+        }
+      });
+    } else if (isSharedValue(value)) {
+      inlineProps[key] = value;
+    }
+  }
+
+  return inlineProps;
+}
+
+export class InlinePropManager {
+  _inlinePropsViewDescriptors: ViewDescriptorsSet | null = null;
+  _inlinePropsMapperId: number | null = null;
+  _inlineProps: StyleProps = {};
+
+  public static hasInlineStyles(style: StyleProps): boolean {
+    if (!style) {
+      return false;
+    }
+    return Object.keys(style).some((key) => {
+      const styleValue = style[key];
+      return (
+        isSharedValue(styleValue) ||
+        (key === 'transform' && isInlineStyleTransform(styleValue))
+      );
+    });
+  }
+
+  public static getInlineStyle(
+    style: Record<string, unknown>,
+    isFirstRender: boolean
+  ) {
+    if (isFirstRender) {
+      return getInlinePropsUpdate(style);
+    }
+    const newStyle: StyleProps = {};
+    for (const [key, styleValue] of Object.entries(style)) {
+      if (
+        !isSharedValue(styleValue) &&
+        !(key === 'transform' && isInlineStyleTransform(styleValue))
+      ) {
+        newStyle[key] = styleValue;
+      }
+    }
+    return newStyle;
+  }
+
+  public attachInlineProps(
+    animatedComponent: React.Component<unknown, unknown>,
+    viewInfo: any // viewTag, viewName, shadowNodeWrapper, viewConfig
+  ) {
+    const newInlineProps: Record<string, any> = extractSharedValuesMapFromProps(
+      animatedComponent.props
+    );
+    const hasChanged = inlinePropsHasChanged(newInlineProps, this._inlineProps);
+
+    if (hasChanged) {
+      if (!this._inlinePropsViewDescriptors) {
+        this._inlinePropsViewDescriptors = makeViewDescriptorsSet();
+
+        const { viewTag, viewName, shadowNodeWrapper, viewConfig } = viewInfo;
+
+        if (Object.keys(newInlineProps).length && viewConfig) {
+          adaptViewConfig(viewConfig);
+        }
+
+        this._inlinePropsViewDescriptors.add({
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          tag: viewTag as number,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          name: viewName!,
+          // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+          shadowNodeWrapper: shadowNodeWrapper!,
+        });
+      }
+      const sharableViewDescriptors =
```

Comment:
```suggestion
      const shareableViewDescriptors =
```

Reply:
It turned out to be quite a common typo in our codebase